code/R3/R3.Core/Math/Infinity.cs

```diff
@@ -13,15 +13,18 @@ public static class Infinity
 		public const double FiniteScale = 10000;
 		public const double InfiniteScale = 500000;
 
+		public static bool IsFinite( double input )
```

Comment:
I like your changes to this file, but I want to make sure NaNs will behave properly.  If IsFinite returns true for them, it will have unintended consequences.  Can you verify and add a comment that we intentially want NaNs to behave a certain way?